pkg/provider/provider_test.go

```diff
@@ -0,0 +1,839 @@
+package provider_test
+
+import (
+	"io/ioutil"
+	"testing"
+
+	mk8s "github.com/containous/maesh/pkg/k8s"
+	"github.com/containous/maesh/pkg/provider"
+	"github.com/containous/maesh/pkg/topology"
+	"github.com/containous/traefik/v2/pkg/config/dynamic"
+	spec "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/specs/v1alpha1"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type TopologyBuilderMock func() (*topology.Topology, error)
+
+func (m TopologyBuilderMock) Build(_ mk8s.IgnoreWrapper) (*topology.Topology, error) {
```

Comment:
Can be a simple function if instead of having a `MiddlewareBuilder` interface we just define a MiddlewareBuilder as a `func(mk8s.IgnoreWrapper) (*topology.Topology, error)`

Reply:
Done, I replace the `MiddlewareBuilder` by a function.